substrate/client/cli/src/commands/export_chain_spec_cmd.rs

```diff
@@ -0,0 +1,63 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use clap::Parser;
+use sc_service::{chain_ops, ChainSpec};
+use std::{
+    fs,
+    io::{self, Write},
+    path::PathBuf,
+};
+
+use crate::error::{Error, Result};
+
+/// Export the embedded chain-spec to a JSON file.
+///
+/// This command loads the embedded chain-spec (for example, when you pass
+/// `--chain /full/path/to/asset-hub-rococo`) and exports it to a JSON file. If `--output`
+/// is not provided, the JSON is printed to stdout.
+#[derive(Debug, Clone, Parser)]
+pub struct ExportChainSpecCmd {
+    /// The chain spec identifier to export.
+    #[arg(long, default_value = "local")]
+    pub chain: String,
```

Comment:
I think we need to have a `shared_params: SharedParams` field instead here. It is part of the `CliConfiguration` trait, and it is used across the board. We should follow the `build-spec` example of using `SharedParams`, and get the `--chain` argument from there.

Reply:
Looking better at this, I realize we can still use `chain`. See context here: https://github.com/paritytech/polkadot-sdk/pull/7719#discussion_r1983019590. Sorry for asking to revert my suggestion 🙈